transaction/src/manifest/enums.rs

```diff
@@ -0,0 +1,137 @@
+use lazy_static::lazy_static;
+use sbor::rust::collections::HashMap;
+
+macro_rules! known_enum {
+    ($map: expr, enum $name:ident { $($variant:ident = $id:expr;)* }) => {
+        $(
+            $map.insert(concat!(stringify!($name), "::", stringify!($variant)), $id);
+        )*
+    };
+}
+
+// TODO: we need a final sanity check before mainnet launch!
+
+lazy_static! {
+    pub static ref KNOWN_ENUM_DISCRIMINATORS: HashMap<&'static str, u8> = {
+        let mut m = HashMap::new();
+
+        // Protocol Buffer syntax
+
+        known_enum!(
+            m,
+            enum AccessRuleEntry {
```

Comment:
Maintaining the parity between what's enclosed in compiler and our actual model is going to be a pain, if they are not auto-generated.  I'd propose we keep it for now as Josh has the idea of making it "data" rather than code.

_(`num_enum` doesn't have any example for variants with fields; I doubt it will work properly)_

Reply:
I'd suggest we instead use a Schema Aggregator to calculate these dynamically from the schema itself? That way it updates automatically? What do you think?

Reply:
Yeah, either with macros on the original struct or deriving from the generated schema. Importantly, we would also need some hardcoded ground truth in e2e tests to detect any change that may invalidate old versions.

Reply:
Yeah, for sure. I suggest some sort of hash of all of them -- or maybe it's better to write it out explicitly here and then compare against the schema in the tests :)

Deriving from the generated schema should be really easy I hope 👍 . I can put my money in my mouth and try it out as part of the schema integration.